MCAntiMalware-Core/src/main/java/optic_fusion1/antimalware/utils/StringUtils.java

```diff
@@ -147,4 +150,20 @@ public static int countMatches(String str, String sub) {
         return count;
     }
 
+    public static String format(String template, Map<String, Object> parameters) {
+        StringBuilder newTemplate = new StringBuilder(template);
+        List<Object> valueList = new ArrayList<>();
+        Matcher matcher = Pattern.compile("[$][{]([\\w-_]+)}").matcher(template);
```

Comment:
There is no reason to have to re-compile this on every method call, convert the Pattern to a CONSTANT.

Reply:
Should be resolved.